app/src/main/java/app/gamenative/ui/screen/settings/DriverManagerDialog.kt

```diff
@@ -0,0 +1,128 @@
+        }
+    }
+
+    AlertDialog(
+        onDismissRequest = onDismiss,
+        title = { Text(text = "Driver Manager") },
+        text = {
+            Column {
+                Text(text = "Import a custom Graphics driver package")
+                lastMessage?.let { Text(text = it, modifier = Modifier.padding(top = 8.dp)) }
+
+                if (installedDrivers.isNotEmpty()) {
+                    Text(text = "\nInstalled custom drivers:")
+                    for (id in installedDrivers) {
+                        Row(modifier = Modifier.fillMaxWidth().padding(top = 6.dp)) {
+                            Text(text = id, modifier = Modifier.weight(1f))
+                            IconButton(onClick = { driverToDelete = id }) {
+                                Icon(
+                                    imageVector = Icons.Filled.Delete,
+                                    contentDescription = "Delete",
+                                )
+                            }
+                        }
+                    }
+                    // Confirmation dialog for deletion
+                    driverToDelete?.let { id ->
+                        AlertDialog(
+                            onDismissRequest = { driverToDelete = null },
+                            title = { Text(text = "Confirm Delete") },
+                            text = { Text(text = "Are you sure you want to remove driver '$id'? This cannot be undone.") },
+                            confirmButton = {
+                                Button(onClick = {
+                                    try {
+                                        val ok = FileUtils.deleteInstalledAdrenoDriver(ctx, id)
+                                        if (ok) {
+                                            lastMessage = "Removed driver: $id"
+                                            refreshDriverList()
+                                        } else {
+                                            lastMessage = "Failed to remove $id"
+                                        }
+                                    } catch (e: Exception) {
+                                        lastMessage = "Error removing $id: ${e.message}"
+                                    }
+                                    driverToDelete = null
+                                }) { Text(text = "Delete") }
+                            },
+                            dismissButton = {
+                                Button(onClick = { driverToDelete = null }) { Text(text = "Cancel") }
+                            },
+                        )
+                    }
+                }
+            }
+        },
+        confirmButton = {
+            Button(onClick = { launcher.launch(arrayOf("application/zip", "*/*")) }) { Text("Import ZIP") }
+        },
+        dismissButton = { Button(onClick = onDismiss) { Text("Close") } },
+    )
+}
+
+private fun handlePickedUri(context: Context, uri: Uri): String {
```

Comment:
For URI-based, I'd say we need to put in a disclaimer that pulling in drives from a URL is their own risk etc.